File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    let stack = []
    for (let num of tokens) {
        let ans = 0;
        if (num !== "+" &&
            num !== "-" &&
            num !== "*" &&
            num !== "/") {
                stack.push(num)
        } else {
            if (num === "+") {
                ans = parseInt(stack[stack.length - 2]) + parseInt(stack[stack.length - 1])
                stack.pop()
                stack.pop()
                stack.push(ans)
            } else if (num === "-") {
                ans = parseInt(stack[stack.length - 2]) - parseInt(stack[stack.length - 1])
                stack.pop()
                stack.pop()
                stack.push(ans)
            } else if (num === "/") {
                ans = Math.trunc(parseInt(stack[stack.length - 2]) / parseInt(stack[stack.length - 1]))
                stack.pop()
                stack.pop()
                stack.push(ans)
            } else {
                ans = parseInt(stack[stack.length - 2]) * parseInt(stack[stack.length - 1])
                stack.pop()
                stack.pop()
                stack.push(ans)
            }
        }
    }
    return stack[0]
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        # array of strings tokens[str]
        # Evaluate it and return an integer
        valid_operators = "+-*/"
        stack = []

        def calculate_expression(num1, num2, operator):
            num1 = int(num1)
            num2 = int(num2)
            if operator == "+":
                return num1 + num2
            elif operator == "-":
                return num1 - num2
            elif operator == "*":
                return num1 * num2
            elif operator == "/":
                return int(num1 / num2)


        for i,num in enumerate(tokens):
            if num in valid_operators:
                op2 = stack.pop()
                op1 = stack.pop()
                res = calculate_expression(op1,op2, num)
                stack.append(res)
            else:
                stack.append(num)

        return int(stack[0])
        